src/morphology/english/getNounForms.js

```diff
@@ -14,58 +12,27 @@ const flatten = require( "lodash/flatten" );
  * Checks if the input word is a possessive form (e.g., "boy's" in "the boy's car") and returns true if that is the case.
  *
  * @param {string} word The word for which to determine if it's a possessive.
+ * @param {Array} possessiveToBaseRegex An array of regex-based rules to bring possessives to base.
  *
  * @returns {boolean} Whether the input word is a possessive form or not.
  */
-const checkPossessive = function( word ) {
+const checkPossessive = function( word, possessiveToBaseRegex ) {
```

Comment:
Since you're not passing a regex, but an array of regex, I think `possessiveToBaseRegexes` would be a better name.